### ISSUE TITLE
use builds from staging build service?

### ISSUE THREAD

**Issue by plata** [PhoenicisOrg/WineBuild#6]
Reasons:
* building our own Wine version causes some errors e.g.: #4
* maintaining this produces work which obviously nobody is willing to do, e.g. PhoenicisOrg/POL-POM-5#1138

At least for OSX, using builds from the build service directly seems to work (see report by @Candunc in PhoenicisOrg/POL-POM-5#1138).

Open questions:
* What about mono/gecko (see also #5)?
* Can the build service build portable .tar.gz for Linux like for OSX?
* Can the build service provide a machine readable file which contains the download links to the released Wine versions (stable, develop, staging)?

**Comment by plata** [comment]
@alesliehughes do you know the answers for the following open question?

> * Can the build service build portable .tar.gz for Linux like for OSX?

> * Can the build service provide a machine readable file which contains the download links to the released Wine versions (stable, develop, staging)?

**Comment by ImperatorS79** [comment]
For #4, shouldn't it be solved simply by installing the related package in the environment currently building POL's wine ? (I have no idea how it is working btw)

**Comment by alesliehughes** [comment]
In short, I don't know, what the build system is able to do.  I don't control anything besides requesting a build to commence.

**Comment by plata** [comment]
@ImperatorS79 yes, but on there are 2 problems with that approach:
1. who does it (I cannot for example)?
2. we cannot ensure that this is the only difference compared to the official Wine builds

**Comment by plata** [comment]
@alesliehughes does this mean that nobody is actually maintaining/responsible for the builder? All the links are still pointing to the old wine-staging.com. That would be really bad because it means that the builder could suddenly stop working and nobody could do anything about it (we do not even have the code of it).

The code could be in
https://github.com/wine-compholio/wine-packaging
https://github.com/wine-compholio/wine-build-tools
but I'm not sure about that.

@slackner @mmueller2012 can you provide more information?

**Comment by ImperatorS79** [comment]
I agree, but for short term purposes, something has to be done with the current builder (which seems to be working) for the people currently using POL/POM. And I assume it's @qparis than can easily fix #4.

**Comment by alesliehughes** [comment]
The builder is being updated and maintained by @mmueller2012.  For me it's a black box, and I'm find with that.

**Comment by plata** [comment]
@ImperatorS79 yes, @qparis should be able to do it. We probably should also update the staging repository.

**Comment by mmueller2012** [comment]
As already mentioned by @alesliehughes the Wine build server is still maintained. I guess the remaining questions also target the Wine build server even though I do not see the reason why they are asked here instead of using wine-devel or the builds@wine-staging.com email address (see https://wiki.winehq.org/Build_Server).

1. Gecko / Mono are separat projects, which need a Windows environment for building. I am not even 100% sure if their build process is properly documented. They are built and pushed to Winehq by their corresponding maintainers.
2. Creating portable builds for Linux is extremely difficult. Unlike macOS, which provides a fixed set of backwards compatible system libraries, on Linux we can only rely on having some kind of libc. You would therefore need to ship a half linux multiarch system as part of the Wine package to achieve compatibility with many distros. This means either using something like snap images and take the libraries from an existing distro or maintain build scripts for like 100 libraries. I am not inclined to use any of the both options, especially since Wine is quite fragile when a library is missing or incompatible. In the best case you get a warning that library could not be loaded and in the worst case the user just runs into a hard to debug problem. I therefore prefer tailoring the Wine builds to the distribution it is supposed to run on.
3. Except for macOS, we provide machine readable files with the list of available wine packages in the form of repositories.

**Comment by plata** [comment]
Thanks for the explanation.

@qparis regarding 2.: How does it even work for PoL/PoM currently?
